experiment/coverage_utils.py

```diff
@@ -0,0 +1,115 @@
+
+
+def merge_profdata_files(src_files, dst_file):
+    """Merge profdata files from |src_files| to |dst_files|."""
+    command = ['llvm-profdata', 'merge', '-sparse'
+              ] + src_files + ['-o', dst_file]
+    result = new_process.execute(command, expect_zero=False)
+    if result.retcode != 0:
+        logger.error('Profdata files merging failed.')
+
+
+def fetch_source_files(benchmarks, report_dir):
+    """Fetch source files to |report_dir|."""
+    for benchmark in benchmarks:
+        coverage_binaries_dir = build_utils.get_coverage_binaries_dir()
+        src_path = os.path.join(coverage_binaries_dir, benchmark, 'src-files')
+        dst_dir = os.path.join(report_dir, benchmark, 'src-files')
+        filesystem.copytree(src_path, dst_dir)
+
+
+def get_profdata_files(experiment, benchmarks, fuzzers, report_dir):
+    """Get profdata files to |report_dir|."""
+    for benchmark in benchmarks:
+        for fuzzer in fuzzers:
+            trial_ids = measurer.get_trial_ids(experiment, fuzzer, benchmark)
+            files_to_merge = [
+                get_profdata_file_path(fuzzer, benchmark, trial_id)
+                for trial_id in trial_ids
+            ]
+            dst_dir = os.path.join(report_dir, benchmark, fuzzer)
+            filesystem.create_directory(dst_dir)
+            dst_file = os.path.join(dst_dir, 'merged.profdata')
+            merge_profdata_files(files_to_merge, dst_file)
+
+
+def fetch_binary_files(benchmarks, report_dir):
+    """Get binary file for each benchmark."""
+    for benchmark in benchmarks:
+        src_file = measurer.get_coverage_binary(benchmark)
+        dst_dir = os.path.join(report_dir, benchmark)
+        filesystem.copy(src_file, dst_dir)
+
+
+def generate_cov_reports(benchmarks, fuzzers, report_dir):
+    """Generate coverage reports for each benchmark and fuzzer."""
+    logger.info('Start generating coverage report for benchmarks.')
+    with multiprocessing.Pool() as pool:
+        generate_cov_report_args = [(benchmark, fuzzer, report_dir)
+                                    for benchmark in benchmarks
+                                    for fuzzer in fuzzers]
+        pool.starmap_async(generate_cov_report, generate_cov_report_args)
+        pool.close()
+        pool.join()
+    logger.info('Finished generating coverage report.')
+
+
+def generate_cov_report(benchmark, fuzzer, report_dir):
+    """Generate the coverage report for one pair of benchmark and fuzzer."""
+    logger.info('Generating coverage report for benchmark:{benchmark} \
+                fuzzer:{fuzzer}.'.format(benchmark=benchmark, fuzzer=fuzzer))
+    dst_dir = os.path.join(report_dir, benchmark, fuzzer)
+    fuzz_target = benchmark_utils.get_fuzz_target(benchmark)
+    profdata_file_path = os.path.join(dst_dir, 'merged.profdata')
+    binary_file_path = os.path.join(report_dir, benchmark, fuzz_target)
+    source_file_path_prefix = os.path.join(report_dir, benchmark, 'src-files')
```

Comment:
This module seems to reimplement a lot of the same logic multiple times. Instead you should have helper functions for things like getting the path to source files for a particular benchmark so there can be one source of truth.